lekce/3/javascript.md

```diff
@@ -1,73 +1,299 @@
-# Javascript 
+# Základy JavaScriptu
 
-v prohlížeči
+----
+
+Už umíš vytvořit _statické_ stránky pomocí HTML a CSS. V minulé lekci jsi se navíc dověděla, že nemusíš psát všechno HTML ručně a ukládat do souborů, ale že jej lze _dynamicky_ generovat na serveru, třeba zrovna Pythonem.
+
+Programování webových stránek se však může odehrávat nejen na straně serveru, ale i přímo u návštěvníka stránek, tedy v jeho prohlížeči. Jen jeden jazyk umí jet přímo v prohlížeči a měnit stránku u uživatele bez toho, aby bylo potřeba ji znovu načíst ze serveru. Tím jazykem je JavaScript a pro webovou tvorbu je velmi důležitý. V této lekci se naučíš jeho základy.
+
+---
 
-cilem je hlavně měnit DOM 
+## Backend nebo frontend?
 
 ----
 
-## Javascript versus Python
+Python jede na serveru, generuje stránky, a posílá je uživateli do prohlížeče. Jakmile je hotové odešle, už je nijak změnit neumí. Proto se jedná o _backend_. Jako _backend_ se také označuje i všechno ostatní, co jede na serveru a uživatel to nemůže přímo vidět, jako např. databáze.
 
-Klíčový pro rychlý pochopení, nechcem nějak moc programovat, takže ano, pochopit jak se dělaj proměnný a že středníky…
+HTML, CSS a JavaScript se interpretují v prohlížeči u uživatele, takže se jedná o _frontend_.
 
 ---
 
-## DOM
+## Co je v JavaScriptu možné?
 
 ----
 
-TBD
+- reagovat na to, že uživatel pohne myší, posouvá stránku, označí text.
+- jakkoliv měnit strukturu HTML a CSS vlastnosti dané stránky:
+    - animovat věci na stránce
+    - posouvat věci, skrývat, odkrývat, přebarvovat
+    - zmenšovat nebo zvětšovat prvky na stránce
+    - přidávat nové HTML i s textem, přepisovat existující
+- přistupovat k informacím, které poskytuje prohlížeč:
+    - datum a čas u uživatele
+    - jestli je uživatel ve tmě nebo na světle, kolik má baterky
+    - ovládat audio a video na stránce
+    - jakým směrem je natočeno jeho zařízení, GPS pozice
+- získávat informace ze serveru i bez nového načtení stránky
 
 ---
 
-## Jak něco v DOMu vybrat 
+## JavaScript vs. Python
 
 ----
 
-TBD
+Jednoduchý program v JavaScriptu a ekvivalentní program v Pythonu mohou vypadat třeba takto:
+
+```js
+function isUkulele(stringsCount) {
+    if (stringsCount == 4) {
```

Comment:
To mě napadlo, ale tohle bude fungovat taky a bude to o jednu odlišnost méně od Pythonu, který znají. Otázka je, zda je rovnou učit správně JavaScript, nebo se snažit do začátku jim neplést hlavu (když uvidí tři rovnítka, budou chtít vysvětlit, proč to tak je a to je potom rabbit hole). Každé má své výhody 🤔